### ISSUE TITLE
Build and create a simple object executes for a very long time

### ISSUE THREAD

**Issue by srgoapp** [AutoFixture/AutoFixture#647]
Example:

We have two classes:

``` c#
public class Class2
{
    public Guid Id { get; set; }
    public Guid ExternalId { get; set; }
    public DateTime ValidFrom { get; set; }
    public DateTime ValidTo { get; set; }
    public string MeteringPointIdentification { get; set; }
    public string InstallationNo { get; set; }
}

public class Class1
{
    public DateTime ValidFrom { get; set; }
    public string Field1 { get; set; }
    public string Field2 { get; set; }
    public string Field3 { get; set; }
    public string Field4 { get; set; }
    public string Field5 { get; set; }
    public string Field6 { get; set; }
    public string Field7 { get; set; }
    public string Field8 { get; set; }
    public string Field9 { get; set; }
    public string Field10 { get; set; }
    public string Field11 { get; set; }
    public string Field12 { get; set; }
    public string Field13 { get; set; }
    public string Field14 { get; set; }
    public string Field15 { get; set; }
    public string Field16 { get; set; }
}
```

We customize Class1:

``` c#
var fixture = new Fixture();
fixture.Customize<Class1>(
    x => x
        .With(mpg => mpg.ValidFrom, DateTime.MinValue)
        .With(mpg => mpg.Field1, string.Empty)
        .With(mpg => mpg.Field2, string.Empty)
        .With(mpg => mpg.Field3, string.Empty)
        .With(mpg => mpg.Field4, string.Empty)
        .With(mpg => mpg.Field5, string.Empty)
        .With(mpg => mpg.Field6, string.Empty)
        .With(mpg => mpg.Field7, string.Empty)
        .With(mpg => mpg.Field8, string.Empty)
        .With(mpg => mpg.Field9, string.Empty)
        .With(mpg => mpg.Field10, string.Empty)
        .With(mpg => mpg.Field11, string.Empty)
        .With(mpg => mpg.Field12, string.Empty)
        .With(mpg => mpg.Field13, string.Empty)
        .With(mpg => mpg.Field14, string.Empty)
        .With(mpg => mpg.Field15, string.Empty)
        .With(mpg => mpg.Field16, string.Empty)
    );
```

We build/create Class2:

``` c#
var class2 = fixture
    .Build<Class2>()
    .With(cmp => cmp.ExternalId, Guid.Empty)
    .With(cmp => cmp.InstallationNo, "ABC") 
    .With(cmp => cmp.MeteringPointIdentification, null)
    .With(cmp => cmp.ValidFrom, DateTime.MinValue)
    .With(cmp => cmp.ValidTo, DateTime.MaxValue)
    .Create();
```

The build/create will take more than 2 minutes execute. If we remove the previos Customize of Class1, the build/create will execute in 140 ms.

If we add an extra field to Class1 and the Cusomize of Class1, the build/create will execute in double time (> 4 minutes).

**Comment by ploeh** [comment]
I can reproduce the issue, but I don't understand why it happens.

As a workaround, could you use `OmitAutoProperties` instead?

```C#
fixture.Customize<Class1>(c => c.OmitAutoProperties());
```

That's not _entirely_ equivalent to the above customization, because it isn't going to set the `string` properties to `string.Empty` - instead, it's not going to set them at all.

It does, however, execute in sub-second time.

**Comment by ploeh** [comment]
Does that workaround resolve your issue?

**Comment by nelsonwellswku** [comment]
I am having this same issue. I am creating a unit test that populates a deep object graph composed completely of auto properties and then maps that deep object graph to another similar structure. Omitting auto property generation isn't really an option for me since the very thing I'm testing depends on it.

I will take a look this weekend and see if I can figure out why it is happening, but if you haven't been able to understand why yet then I don't know how much help I can be :(

**Comment by Zvirja** [comment]
@srgoapp, @nelsonwellswku If the issue still happens with the aforementioned scenario, could you please profile code execution using the [dotTrace](https://www.jetbrains.com/profiler/) profiler in __tracing__ mode? Take 2 samples: without `Customize()` invocation that causes issues and with it (so we could compare graphs).

Analysis of the profiling snapshots should simplify the investigation much.

**Comment by zvirja** [comment]
@srgoapp, @nelsonwellswku Could you please find some time to capture the snapshots if possible? 😳

**Comment by zvirja** [comment]
This has been fixed in [v3.50.5](https://github.com/AutoFixture/AutoFixture/releases/tag/v3.50.5). Now it takes about 130 ms on my machine (previously about 3 mins).
